pkg/sql/logictest/tests/local/local_test: TestLogic_notice failed

pkg/sql/logictest/tests/local/local_test.TestLogic_notice [failed](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/6347318?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/6347318?buildTab=artifacts#/) on master @ [e5917605f751c569ddd90062250a40cafd7ccf30](https://github.com/cockroachdb/cockroach/commits/e5917605f751c569ddd90062250a40cafd7ccf30):


```
=== RUN   TestLogic_notice
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e97884aff093cbce900d7154b4fd7ebb/logTestLogic_notice1433902353
    test_log_scope.go:79: use -show-logs to present logs inline
[05:12:59] setting distsql_workmem='91155B';
[05:12:59] rng seed: 2633035918254106233

=== CONT  TestLogic_notice
    logic.go:3920: -- test log scope end --
--- FAIL: TestLogic_notice (2.04s)
=== RUN   TestLogic_notice/test_notice_severity
    logic.go:2743: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3581/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/notice:56: UNLISTEN temp
        expected:
            NOTICE: unimplemented: CRDB does not support LISTEN, making UNLISTEN a no-op
            HINT: You have attempted to use a feature that is not yet implemented.
            See: https://go.crdb.dev/issue-v/41522/v22.2
        but found (query options: "noticetrace") :
            NOTICE: unimplemented: CRDB does not support LISTEN, making UNLISTEN a no-op
            HINT: You have attempted to use a feature that is not yet implemented.
            See: https://go.crdb.dev/issue-v/41522/dev
[05:12:59] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3581/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/notice with config local: 12 tests, 1 failures
--- total: 12 tests, 1 failures
    --- FAIL: TestLogic_notice/test_notice_severity (0.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_notice.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-19301

pkg/sql/logictest/tests/local/local_test.TestLogic_notice [failed](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/6369522?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/6369522?buildTab=artifacts#/) on master @ [2372698da1dfacb90f60c6a63f2c1298d1db16b8](https://github.com/cockroachdb/cockroach/commits/2372698da1dfacb90f60c6a63f2c1298d1db16b8):


```
=== RUN   TestLogic_notice
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e97884aff093cbce900d7154b4fd7ebb/logTestLogic_notice2354176572
    test_log_scope.go:79: use -show-logs to present logs inline
[05:14:04] setting distsql_workmem='92698B';
[05:14:04] rng seed: 8381029383119382039

=== CONT  TestLogic_notice
    logic.go:3920: -- test log scope end --
--- FAIL: TestLogic_notice (1.77s)
=== RUN   TestLogic_notice/test_notice_severity
    logic.go:2743: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3582/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/notice:56: UNLISTEN temp
        expected:
            NOTICE: unimplemented: CRDB does not support LISTEN, making UNLISTEN a no-op
            HINT: You have attempted to use a feature that is not yet implemented.
            See: https://go.crdb.dev/issue-v/41522/v22.2
        but found (query options: "noticetrace") :
            NOTICE: unimplemented: CRDB does not support LISTEN, making UNLISTEN a no-op
            HINT: You have attempted to use a feature that is not yet implemented.
            See: https://go.crdb.dev/issue-v/41522/dev
[05:14:05] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3582/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/notice with config local: 12 tests, 1 failures
--- total: 12 tests, 1 failures
    --- FAIL: TestLogic_notice/test_notice_severity (0.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_notice.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -3413,7 +3413,16 @@ func (t *logicTest) finishExecQuery(query logicQuery, rows *gosql.Rows, err erro
 		}
 		for i := range query.expectedResults {
 			expected, actual := query.expectedResults[i], actualResults[i]
-			resultMatches := expected == actual
+			var resultMatches bool
+			if query.noticetrace {
+				resultMatches, err = regexp.MatchString(expected, actual)
+				if err != nil {
+					return errors.CombineErrors(makeError(), err)
+				}
+			} else {
+				resultMatches = expected == actual
+			}
+
 			// Results are flattened into columns for each row.
 			// To find the coltype for the given result, mod the result number
 			// by the number of coltypes.

--- a/pkg/sql/logictest/testdata/logic_test/notice
+++ b/pkg/sql/logictest/testdata/logic_test/notice
@@ -57,4 +57,4 @@ UNLISTEN temp
 ----
 NOTICE: unimplemented: CRDB does not support LISTEN, making UNLISTEN a no-op
 HINT: You have attempted to use a feature that is not yet implemented.
-See: https://go.crdb.dev/issue-v/41522/v22.2
+See: https://go.crdb.dev/issue-v/41522/*

